### ISSUE TITLE
preActions for Target Schemes (suggestion)

### ISSUE THREAD

**Issue by abeintopalo** [yonaskolb/XcodeGen#279]
Currently it seems it is not possible to define `preActions` for target schemes.
Consider the following target scheme:

```
    scheme:  
      configVariants:
        - Dev
        - AdHoc
        - AppStore
```
it would be nice to have the ability to specify an (optional) common `preActions` element that is embedded for each generated scheme.

**Comment by yonaskolb** [comment]
Yeah, adding pre and post actions to target scheme will be easy. They will be common to all build types though, if that suits your needs. 
I'm sure you're aware of the more configurable top level schemes that you can add pre and post actions per build type to.
https://github.com/yonaskolb/XcodeGen/blob/master/Docs/ProjectSpec.md#scheme

**Comment by abeintopalo** [comment]
Yeah, I've checked top level schemes but using that I would have to explicitly define each scheme variant per configuration if I understood correctly.

**Comment by yonaskolb** [comment]
That's right. I've created a PR for this here https://github.com/yonaskolb/XcodeGen/pull/280

**Comment by abeintopalo** [comment]
Wow, that was fast. Thank you so much! :)
